### PR TITLE
fix: Deal with pagination of response for modified PR files

### PR DESCRIPTION
**Pull request by pranavrc**
https://github.com/screwdriver-cd/screwdriver/issues/1510

This PR fixes the logic to deal with pagination and fetch all pages of results instead of just the first page with 30 results.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "chai": "^3.5.0",
     "eslint": "^4.19.1",
     "eslint-config-screwdriver": "^3.0.1",
+    "eslint-plugin-import": "^2.16.0",
```
Comment (tkyi) [inline package.json]: shouldn't need this

Comment (pranavrc) [reply package.json]: fixed

File: index.js [hunk 2]
```diff
@@ -960,13 +960,13 @@ class GithubScm extends Scm {
 
             try {
                 const files = await this.breaker.runCommand({
-                    action: 'listFiles',
-                    scopeType: 'pulls',
+                    scopeType: 'paginate',
+                    route: 'GET /repos/:owner/:repo/pulls/:number/files',
                     token,
                     params: { owner, repo, number }
```
Comment (d2lam) [inline index.js]: Not sure if I understand the implementation, it looks like the call is default to paginate https://octokit.github.io/rest.js/#api-Pulls-listFiles. 
We can try to increase `per_page` param to `100`. Or if we want to call ALL of them then we need to make more calls until there is no more data.

Comment (pranavrc) [reply index.js]: The max number of results that github returns is 300. The `paginate` function does the pagination internally, so we don't have to deal with the pagination. https://github.com/octokit/rest.js#pagination

Comment (tkyi) [reply index.js]: According to the documentation, looks like octokit handles pagination with `paginate` and magically will return all the results: https://github.com/octokit/rest.js#pagination

Comment (d2lam) [reply index.js]: oh cool cool! :+1

**Pull request merged**